src/Evolution/Systems/Cce/AnalyticSolutions/LinearizedBondiSachs.cpp

```diff
@@ -0,0 +1,511 @@
+                                   l_max, time);
+}
+
+void LinearizedBondiSachs::linearized_dr_bondi_j(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 2>*> dr_bondi_j,
+    const size_t l_max, const double time) const noexcept {
+  const std::complex<double> dr_j_2_coefficient =
+      -0.5 * sqrt(6.0) *
+      (c_2a_ / square(extraction_radius_) - c_2b_ / pow<4>(extraction_radius_));
+
+  const std::complex<double> dr_j_3_coefficient =
+      2.0 * sqrt(15.0) *
+      (-0.1 * c_3a_ / square(extraction_radius_) +
+       c_3b_ / pow<5>(extraction_radius_) +
+       std::complex<double>(0.0, 1.0) * frequency_ *
+           (0.5 * c_3b_ / (pow<4>(extraction_radius_))));
+
+  assign_components_from_l_factors(dr_bondi_j, dr_j_2_coefficient,
+                                   dr_j_3_coefficient, l_max, time);
+}
+
+void LinearizedBondiSachs::linearized_dr_bondi_u(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 1>*> dr_bondi_u,
+    const size_t l_max, const double time) const noexcept {
+  const std::complex<double> dr_u_2_coefficient =
+      sqrt(6.0) * (-c_2a_ / pow<3>(extraction_radius_) -
+                   c_2b_ / pow<5>(extraction_radius_) +
+                   std::complex<double>(1.0, 0.0) * frequency_ *
+                       (-c_2b_ / (3.0 * square(extraction_radius_))));
+
+  const std::complex<double> dr_u_3_coefficient =
+      sqrt(12.0) *
+      (-c_3a_ / pow<3>(extraction_radius_) +
+       2.0 * square(frequency_) * c_3b_ / (pow<4>(extraction_radius_)) -
+       5.0 * c_3b_ / pow<6>(extraction_radius_) +
+       std::complex<double>(0.0, 1.0) * frequency_ *
+           (-5.0 * c_3b_ / pow<5>(extraction_radius_)));
+
+  assign_components_from_l_factors(dr_bondi_u, dr_u_2_coefficient,
+                                   dr_u_3_coefficient, l_max, time);
+}
+
+void LinearizedBondiSachs::linearized_dr_bondi_w(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 0>*> dr_bondi_w,
+    const size_t l_max, const double time) const noexcept {
+  const std::complex<double> dr_w_2_coefficient =
+      2.0 * square(frequency_) * c_2b_ / pow<3>(extraction_radius_) -
+      2.0 * c_2b_ / pow<5>(extraction_radius_) +
+      std::complex<double>(0.0, 1.0) * frequency_ *
+          (-3.0 * c_2b_ / pow<4>(extraction_radius_));
+
+  const std::complex<double> dr_w_3_coefficient =
+      -10.0 * frequency_ * c_3b_ / pow<5>(extraction_radius_) -
+      15.0 * c_3b_ / pow<6>(extraction_radius_) +
+      std::complex<double>(0.0, 1.0) * frequency_ *
+          (4.0 * square(frequency_) * c_3b_ / pow<3>(extraction_radius_) +
+           12.0 * frequency_ * c_3b_ / pow<4>(extraction_radius_));
+
+  assign_components_from_l_factors(dr_bondi_w, dr_w_2_coefficient,
+                                   dr_w_3_coefficient, l_max, time);
+}
+
+void LinearizedBondiSachs::linearized_dt_bondi_j(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 2>*> dt_bondi_j,
```

Comment:
document that there is a factor of (we don't know) * `i` added in the `assign_dt_components_from_l_factors` call compared to the documentation...

Reply:
Sorry, I'm a little confused - do you mean the `frequency_ * (std::complex<double>(0.0, 1.0) * cos(frequency_ * time) + sin(frequency_ * time))`? That's just equal to $i \nu e^{i \nu t}$ that I think made it into the documentation -- though it is in the wrong place, I'll move it inside the Re() operator. Was there another factor that's documented wrong?

Reply:
there were factors of  `sqrt(2.0)` that we were confused about, that it looks like you've removed...

Reply:
oh, I see - yeah, there were some in the dt version as well that I didn't finish factoring out consistently. I added another fixup to make that all clear and correct. Thanks!